SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.cpp

```diff
@@ -348,11 +409,57 @@ void SSpatialOutputLog::FormatAndPrintRawLogLine(const FString& LogLine)
 		LogVerbosity = ELogVerbosity::Log;
 	}
 
-	// Serialization must be done on the game thread.
-	AsyncTask(ENamedThreads::GameThread, [this, LogMessage, LogVerbosity, LogCategory]
+	AppendOutput(LogMessage, LogVerbosity, LogCategory);
+}
+
+void SSpatialOutputLog::ParseSingleNodeRuntimeLogContent(const FString& Content)
+{
+	TArray<FString> LogLines;
+
+	Content.ParseIntoArray(LogLines, TEXT("\n"), true);
+
+	for (const FString& LogLine : LogLines)
 	{
-		Serialize(*LogMessage, LogVerbosity, FName(*LogCategory));
-	});
+		FormatAndPrintSingleNodeRuntimeLogLine(LogLine);
+	}
+}
+
+void SSpatialOutputLog::FormatAndPrintSingleNodeRuntimeLogLine(const FString& LogLine)
+{
+	const FRegexPattern LogPattern = FRegexPattern(TEXT("\\[[0-9\\-\\.:\\s]*\\] \\[([^\\]]*)\\] (.*)"));
+	FRegexMatcher LogMatcher(LogPattern, LogLine);
+
+	if (!LogMatcher.FindNext())
+	{
+		return;
+	}
+
+	const FString LogLevelText = LogMatcher.GetCaptureGroup(1);
+	const FString LogMessage = LogMatcher.GetCaptureGroup(2);
+
+	ELogVerbosity::Type LogVerbosity;
+	if (LogLevelText.Contains(TEXT("critical")))
+	{
+		LogVerbosity = ELogVerbosity::Fatal;
+	}
+	else if (LogLevelText.Contains(TEXT("error")))
+	{
+		LogVerbosity = ELogVerbosity::Error;
+	}
+	else if (LogLevelText.Contains(TEXT("warning")))
+	{
+		LogVerbosity = ELogVerbosity::Warning;
+	}
+	else if (LogLevelText.Contains(TEXT("info")))
+	{
+		LogVerbosity = ELogVerbosity::Log;
+	}
+	else
+	{
+		LogVerbosity = ELogVerbosity::Verbose;
+	}
+
+	AppendOutput(LogMessage, LogVerbosity, TEXT("Runtime"));
```

Comment:
Nice, I like the single category. Question though, are there runtime log categories present in the runtime logs? Like which logger they came from for example

Reply:
No the new logs don't carry much meta data at all. They just have a timestamp and a verbosity. You can't really tell where each entry was generated from without semantic knowledge about the content.

Reply:
Cool